packages/util-endpoints/src/utils/toDebugString.ts

```diff
@@ -0,0 +1,26 @@
+import { EndpointParameters, EndpointV2 } from "@aws-sdk/types";
+
+import { GetAttrValue } from "../lib";
+import { EndpointObject, FunctionObject, FunctionReturn } from "../types";
+
+export function toDebugString(input: EndpointParameters): string;
+export function toDebugString(input: EndpointV2): string;
+export function toDebugString(input: GetAttrValue): string;
+export function toDebugString(input: FunctionObject): string;
+export function toDebugString(input: FunctionReturn): string;
+export function toDebugString(input: EndpointObject): string;
+export function toDebugString(input: any): string {
```

Comment:
these are function declaration overloads because this allows static analysis to find usages specific to a certain overload, as opposed to

`(arg: A | B | C | D | E) => string`